src/i2p.cpp

```diff
@@ -376,7 +376,7 @@ void Session::CreateIfNotCreatedAlready()
     m_session_id = session_id;
     m_control_sock = std::move(sock);
 
-    LogPrintf("I2P: SAM session created: session id=%s, my address=%s\n", m_session_id,
+    LogPrintf("[i2p] SAM session created: session id=%s, my address=%s\n", m_session_id,
```

Comment:
I understand the idea, but I don't think we should do it this way by inserting `[cat]` manually. We need a way to log things with a category even without `-debug=`.
This is why I have a commit in #25202 to log everything >= Warning unconditionally. I'm not sure that is the best solution but I like it more than this.

Reply:
Thanks, SGTM, will just do the de-dupe then (also for ZMQ and PRUNE).

Reply:
That's fine but to be clear I mean I agree conceptually with doing it, just not in this way.

Reply:
Done.

Reply:
> That's fine but to be clear I mean I agree conceptually with doing it, just not in this way.

Agree, thanks for clarifying.  I'll start with reviewing your change and then go from there / update here.

Reply:
How about `LogPrintfCategory`?
```diff
--- a/src/logging.h
+++ b/src/logging.h
@@ -198,6 +198,7 @@ static inline void LogPrintf_(const std::string& logging_function, const std::st
 
 #define LogPrintf(...) LogPrintLevel_(BCLog::LogFlags::NONE, BCLog::Level::None, __VA_ARGS__)
+#define LogPrintfCategory(category, ...) LogPrintLevel_(category, BCLog::Level::None, __VA_ARGS__)

--- a/src/i2p.cpp
+++ b/src/i2p.cpp
@@ -376,7 +376,7 @@ void Session::CreateIfNotCreatedAlready()
 
-    LogPrintf("I2P: SAM session created: session id=%s, my address=%s\n", m_session_id,
+    LogPrintfCategory(BCLog::I2P, "SAM session created: session id=%s, my address=%s\n", m_session_id,
```

debug log
```
2022-05-25T16:03:02Z [i2p] SAM session created
```

Reply:
Done

Reply:
Yes, lgtm!